### PR TITLE
Update dummy context decorator to match Django atomic

### PR DESCRIPTION
**Pull request by bocribbz**
The dummy context manager now matches the exact same structure as Django's `@atomic` so it works for both `@atomic` and `@atomic()`

### CODE REVIEW

**Comment by caffodian** [comment]
Huh.  That first `using` still gets me, af first glance, even with the comment.

So this is from:
https://github.com/django/django/blob/master/django/db/transaction.py#L276
pretty much.  

Can we add test cases to show that it works with both, syntactically?

File: nosedjango/nosedjango.py [hunk 1]
```diff
@@ -70,8 +70,14 @@ def inner(*args, **kwargs):
         return inner
 
 
-def _dummy_context_manager(*args, **kwargs):
-    return ContextDecorator(*args, **kwargs)
+def _dummy_context_manager(using=None, savepoint=True):
+    # Bare decorator: @atomic -- although the first argument is called
+    # `using`, it's actually the function being decorated.
+    if callable(using):
+        return ContextDecorator(None, savepoint)(using)
```
Comment (caffodian) [inline nosedjango/nosedjango.py]: In hindsight, I think we should rename this to DummyContextDecorator or something.  I keep forgetting that this isn't he contextlib-compatible one

**Comment by michaelgzoller** [comment]
:+1:

File: nosedjangotests/polls/tests/test_dummy_decorator.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+from importlib import import_module
+
+import django
+try:
+    from django.db.transaction import atomic
+except ImportError:
+    from django.db.transaction import commit_on_success as atomic
```
Comment (jlward) [inline nosedjangotests/polls/tests/test_dummy_decorator.py]: These imports aren't actually being used. Is that on purpose?

File: nosedjangotests/polls/tests/transaction_tester.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+try:
+    from django.db.transaction import atomic
+except ImportError:
+    from django.db.transaction import commit_on_success as atomic
```
Comment (jlward) [inline nosedjangotests/polls/tests/transaction_tester.py]: We should also probably have a test using transaction.atomic as well.

File: nosedjangotests/polls/tests/test_dummy_decorator.py (hunk 2 above)
Comment (bocribbz) [inline nosedjangotests/polls/tests/test_dummy_decorator.py]: good find, leftovers from first attempt to apply `@atomic` to a method in this file.

**Comment by jlward** [comment]
Looks fine to me.

**Comment by bocribbz** [comment]
I managed to track down the problem to using `importlib.import_module` to load a file that has `@atomic` applied to its methods, which causes the decorator to take the place of the actual method, thus erroring about not matching kwargs.

**Pull request merged**